design/subdomain-delegation.md

```diff
@@ -0,0 +1,65 @@
+# Subdomain Delegation
+
+_Status_: Draft, in review.
+
+This document outlines an addition to the [IngressRoute][0] specification to permit subdomains to inherit domain wide parameters such as wildcard TLS certificates without exposing that configuration to the specific subdomain.
+
+## Goals
+
+- Permit a number of subdomains to be served using a wildcard TLS certificates without requiring the IngressRoute record for that subdomain to be a part of the namespace holding the TLS wildcard certificate's Secret.
+
+## Non-goals
+
+- To support hosting a Kubernetes service for a wildcard domain (eg, \*.whatever.net) itself. 
+
+## Background
+
+Currently the Secret containing the TLS certificate must be co-located in the same namespace as the Ingress or root IngressRoute object referencing that secret.
+This requirement complicates deployment patterns where wildcard TLS certificates are used, specifically the use of a wildcard certificate to secure a number of subdomains where the Ingress/IngressRoute records for those subdomains do not share the same namespace as the wildcard TLS certificate.
+For example, presenting foo.example.com using the certificate for \*.example.com when the Ingress/IngressRoute object for foo.example.com does not share the same namespace as the Secret containing \*.example.com.
+
+This proposal introduces a new type of delegation, subdomain delegation, which permits the root IngressRoute object to define a TLS certificate to be used for a set of IngressRoute resources denoted as subdomains.
+
+## High-Level Design
+
+- `spec.virtualhost.fqdn` can be treated as a domain _suffix_, if the `spec.delegate` key is present.
+- `spec.virtualhost.tls.secretName` indicates the certificate to be applied to all subdomains delegated from this domain root.
+
+## Detailed Design
+
+This proposal extends the IngressRoute specification to permit delegation of subdomains.
+
+```
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata:
+  name: domain-root
+  namespace: mycompany
+spec:
+  virtualhost:
+    fqdn: .mycompany.com
```

Comment:
Is `foo.bar.mycompany.com` an acceptable subdomain?  I wanted to be clear on the definition of `suffix` in this context

Reply:
That's a good point. Practically if this feature is used to delegate TLS subdomains it can only really be used for a single level. But from the POV of a data model it seems weird to add that restriction.

Reply:
I think it's also interesting to work through chained delegations like we can do in IngressRoute